cli/http.go

```diff
@@ -0,0 +1,91 @@
+package cli
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+var (
+	dcosAuthToken string
+	dcosUrl       string
+	serviceName   string
+)
+
+func HTTPGet(urlPath string) *http.Response {
+	return HTTPQuery("GET", urlPath, "", true)
+}
+
+func HTTPDelete(urlPath string) *http.Response {
+	return HTTPQuery("DELETE", urlPath, "", true)
+}
+
+func HTTPPost(urlPath string, jsonPayload string) *http.Response {
+	return HTTPQuery("POST", urlPath, jsonPayload, true)
+}
+
+func HTTPPut(urlPath string, jsonPayload string) *http.Response {
+	return HTTPQuery("PUT", urlPath, jsonPayload, true)
+}
+
+func HTTPQuery(method string, urlPath string, jsonString string, statusCheck bool) *http.Response {
+	request := CreateHTTPRequest(method, urlPath, jsonString)
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		// err contains method/parsedUrl. avoid repeating ourselves:
+		log.Fatalf("HTTP Query failed: %s\n"+
+			"- Is 'core.dcos_url' set correctly? Check 'dcos config show core.dcos_url'\n"+
+			"- Is 'core.dcos_acs_token' set correctly? Call 'dcos auth login' to log in.", err)
+	}
+	if response.StatusCode == 401 {
+		log.Fatalf("Got 401 Unauthorized response from %s. Bad auth token? Call 'dcos auth login' to log in.", request.URL)
+	}
+	if statusCheck &&
+		(response.StatusCode < 200 || response.StatusCode >= 300) {
+		log.Fatalf("HTTP %s Query for %s failed: %s", method, request.URL, response.Status)
+	}
+	if Verbose {
+		log.Printf("Response: %s (%d bytes)", response.Status, response.ContentLength)
+	}
+	return response
+}
+
+func CreateHTTPRequest(method string, urlPath string, jsonString string) *http.Request {
```

Comment:
I'm not a big fan of having to rewrite the components of the dcos cli that we need, to get around the lack of a `dcos` python module.  I realize we have to do that if we're writing in go, but still, kinda sucks...